src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -16019,33 +16019,33 @@ instruct cmpFastUnlock(rFlagsReg cr, iRegP object, iRegP box, iRegPNoSp tmp, iRe
   ins_pipe(pipe_serial);
 %}
 
-instruct cmpFastLockLightweight(rFlagsReg cr, iRegP object, iRegP box, iRegPNoSp tmp, iRegPNoSp tmp2)
+instruct cmpFastLockLightweight(rFlagsReg cr, iRegP object, iRegP box, iRegPNoSp tmp, iRegPNoSp tmp2, iRegPNoSp tmp3)
 %{
   predicate(LockingMode == LM_LIGHTWEIGHT);
   match(Set cr (FastLock object box));
-  effect(TEMP tmp, TEMP tmp2);
+  effect(TEMP tmp, TEMP tmp2, TEMP tmp3);
```

Comment:
Why not use `box` as the temp instead of intruducing a separate temp?
```suggestion
  effect(TEMP tmp, TEMP tmp2, TEMP box);
```

Reply:
Making an input TEMP will crash inside C2 when doing register allocation. 
```
assert(opcnt < numopnds) failed: Accessing non-existent operand

V  [libjvm.dylib+0xcc650c]  MachNode::in_RegMask(unsigned int) const+0x1f0
V  [libjvm.dylib+0x3d136c]  PhaseChaitin::gather_lrg_masks(bool)+0x1130
V  [libjvm.dylib+0x3cef9c]  PhaseChaitin::Register_Allocate()+0x150
V  [libjvm.dylib+0x4c3860]  Compile::Code_Gen()+0x1f4
V  [libjvm.dylib+0x4c17a4]  Compile::Compile(ciEnv*, ciMethod*, int, Options, DirectiveSet*)+0x1388
V  [libjvm.dylib+0x38abf0]  C2Compiler::compile_method(ciEnv*, ciMethod*, int, bool, DirectiveSet*)+0x1e0
V  [libjvm.dylib+0x4df028]  CompileBroker::invoke_compiler_on_method(CompileTask*)+0x854
V  [libjvm.dylib+0x4de46c]  CompileBroker::compiler_thread_loop()+0x348
V  [libjvm.dylib+0x8c10e4]  JavaThread::thread_main_inner()+0x1dc
V  [libjvm.dylib+0x117f7f8]  Thread::call_run()+0xf4
V  [libjvm.dylib+0xe53724]  thread_native_entry(Thread*)+0x138
C  [libsystem_pthread.dylib+0x7034]  _pthread_start+0x88
```

Maybe this can be resolved and support for TEMP input registers can be added.